### PR TITLE
Fix/factory grace note

### PR DESCRIPTION
**Pull request by rvilarl**
TODO: Factory.GraceNote  resolved.
TODO: Stroke constructor 2nd parameter resolved.

### CODE REVIEW

File: src/gracenote.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export class GraceNote extends StaveNote {
     return 0.66;
   }
 
-  constructor(note_struct: GraceNoteStruct) {
+  constructor(note_struct: Partial<GraceNoteStruct>) {
```
Comment (ronyeh) [inline src/gracenote.ts:27]: Sometimes I see note_struct and sometimes it is noteStruct. Should we standardize to `noteStruct` where possible? Maybe the underscores should be reserved for keys of parameter objects, or legacy public fields.

File: src/stemmablenote.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ export abstract class StemmableNote extends Note {
   protected flag?: Glyph;
   protected stem_extension_override?: number;
 
-  constructor(note_struct: NoteStruct) {
+  constructor(note_struct: Partial<NoteStruct>) {
```
Comment (ronyeh) [inline src/stemmablenote.ts:25]: note_struct => noteStruct :-)

I figure we might as well rename it now while we are working on this PR. (I see stavenote.ts uses noteStruct).

File: src/gracenote.ts (hunk 1 above)
Comment (rvilarl) [reply src/gracenote.ts:27]: Agree, it will add a few files though

**Comment by 0xfe** [comment]
Thanks for this @rvilarl. Also, thanks for cleaning up the `noteStruct` inconsistencies.

**Pull request merged**